tests/test_engine.py

```diff
@@ -52,6 +52,18 @@ def test_get_lookup_url(self):
         assert final_uri_with_params == '1/2/?extra_param=3'
         SampleResourceModel._lookup_urls = []
 
```

Comment:
What if you used [OrderedDict](https://docs.python.org/2/library/collections.html#collections.OrderedDict) here instead?